src/Core/Checkout/Document/Renderer/CreditNoteRenderer.php

```diff
@@ -106,13 +121,24 @@ public function render(array $operations, Context $context, DocumentRendererConf
                     continue;
                 }
 
-                $lineItems = $order->getLineItems();
-                $creditItems = new OrderLineItemCollection();
+                $liveLineItems = $order->getLineItems() ?? new OrderLineItemCollection();
+                $liveCreditItems = $liveLineItems->filterByType(LineItem::CREDIT_LINE_ITEM_TYPE);
 
-                if ($lineItems) {
-                    $creditItems = $lineItems->filterByType(LineItem::CREDIT_LINE_ITEM_TYPE);
+                if ($liveCreditItems->count() === 0) {
+                    throw DocumentException::generationError(
+                        'Can not generate credit note document because no credit line items exists. OrderId: ' . $operation->getOrderId()
+                    );
                 }
 
+                $referencedInvoiceId = $operation->getReferencedDocumentId();
+                $invoiceCreditIds = $this->getCreditIdsOnInvoiceDocument($referencedInvoiceId);
+                $creditNoteItemIds = $this->getPreviouslyCreditedIdsForInvoice($referencedInvoiceId);
+
+                $creditItems = $liveCreditItems->filter(
+                    fn (OrderLineItemEntity $item) => !\in_array($item->getId(), $invoiceCreditIds, true)
+                        && !\in_array($item->getId(), $creditNoteItemIds, true)
+                );
+
                 if ($creditItems->count() === 0) {
                     throw DocumentException::generationError(
                         'Can not generate credit note document because no credit line items exists. OrderId: ' . $operation->getOrderId()
```

Comment:
```suggestion
                        'Can not generate credit note document because no unprocessed credit line items exists. OrderId: ' . $operation->getOrderId()
```

Not part of the change, but since this error is now shown when credit line items do exist, but all have already been processed in previous credit notes, I think specifying "unprocessed" makes sense here. What do you think?